evaluation_framework/evaluation.py

```diff
@@ -0,0 +1,172 @@
+        self.custom_metrics = config.paths["custom_metrics"]
+        self.max_concurrent = config.max_concurrent
+        self.throttle_value = config.throttle_value
+        self.app = app
+        # Initialize configuration
+        if self.app:
+            self.llm_model = create_model_from_app_config(self.app)
+        else:
+            self.llm_model = create_llm_model()
+
+    async def prepare_eval_data(self, eval_data: List[Dict], rag_system: RAG) -> List[Dict]:
+        """Process goldens and return evaluation data"""
+        # Process all questions in one batch
+        questions = [eval_case["input"] for eval_case in eval_data]
+        answers = await rag_system.ask_questions(questions)
+        # Update eval_data_list with results
+        for i, eval_case in enumerate(eval_data):
+            _answer = answers["results"][i]["message"]["content"]
+            _context = answers["results"][i]["context"]['data_points']["text"]
+            eval_case["actual_output"] = _answer
+            eval_case["retrieval_context"] = _context
+
+        return eval_data
+    
+    def prepare_goldens(self, goldens: List[Dict]) -> List[Dict]:
+        """Prepare goldens for evaluation"""
+        goldens = [g.model_dump() for g in goldens]
+        eval_data_list = []
+        for golden in goldens:
+            eval_data = {
+                "input": golden["input"],
+                "expected_output": golden["expected_output"],
+                "context": golden["context"]
+            }
+            eval_data_list.append(eval_data)
+        return eval_data_list
+    
+    async def _evaluate(self, eval_data: List[Dict]) -> None:
+        """
+        Run evaluation on the provided test cases.
+        eval_data: List of evaluation data
+        """
+        custom_metrics = read_json(self.custom_metrics)
+        metrics = create_metrics(self.llm_model, custom_metrics)
+
+        test_cases = [LLMTestCase(**test_case) for test_case in eval_data]
+
+        # Run evaluation
+        results = evaluate(
+            test_cases=test_cases,
+            metrics=metrics,
+            throttle_value=self.throttle_value,
+            max_concurrent=self.max_concurrent,
+            write_cache=False
+        )
+        # Save results
+        if self.results_path:
+            save_results(results, self.results_path)
+
+        return results
+    
+    def load_test_cases(self, test_file: str) -> List[Dict]:
+        """Load test cases from a JSON file"""
+        test_cases = read_json(test_file)["test_cases"]
+        return test_cases
+    
+    def prepare_tests(self, test_cases: List[Dict]) -> None:
+        """Run tests on the provided test cases"""
+        custom_metrics = read_json(self.custom_metrics)
+        metrics = create_metrics(self.llm_model, custom_metrics)
+        
+        test_cases = [LLMTestCase(**test_case) for test_case in test_cases]
+        return test_cases, metrics
+
+
+def create_llm_model() -> DeepEvalBaseLLM:
```

Comment:
Looks that these functions are only used inside class, perhaps moving into class can improve code organization

Reply:
Fixed